### ISSUE TITLE
Escargot - A simple escrow service.

### ISSUE THREAD

**Issue by okmrponcho** [district0x/district-proposals#67]
Check out Google Doc:
https://docs.google.com/document/d/1Ocyofbsnph4g5ZULsZXQBFWXIZ8vcGHQD_RVYnPl_YU/edit?usp=sharing

![screen shot 2017-07-29 at 8 42 33 pm](https://user-images.githubusercontent.com/29905125/28859004-90a16284-7709-11e7-973c-04526e375058.png)

Escargot is a simple escrow service that enables commerce executed via smart contracts. It connects transacting parties with a market of reputable third party arbiters. Arbiters resolve disputes that arise between transacting parties based on previously established terms and conditions. Arbiters earn a fee when their services are utilized and disputes are successfully resolved. 

## Purpose

Smart contracts are revolutionizing the way business is done, but real interactions between human beings cannot always be resolved in code. Smart contracts allow guaranteed execution of transactions according to predefined terms, but real life is not always black and white. Disputes will continue to arise that require human discretion to resolve. 

**Introducing Escargot.** Escargot is a practical solution to the gray areas that exist in smart contracts. It does this along two pillars. First, it defines a simple escrow-enabled transaction structure that enables a predefined third party arbiter to resolve disputes. Second, it creates a district0x marketplace of reputable arbiters who wish to provide their services. Business transacted in the Escargot marketplace will comply with simple escrow enabled transaction structure, allowing for seamless execution of smart contracts. Disputes are resolved and fees are paid directly in the Escargot marketplace. 

Arbiters are subject to evaluation using a special reputation model. The model encourages all community members to retrospectively evaluate arbiter decisions, which maintains a high level of quality in the market. 

**Es-crow, noun,** A contractual agreement in which monetary disbursements are withheld pending conditions agreed to by the transacting parties

## Functionality

Escargot relies on three underlying technologies:

**1) d0xINFRA** - Escargot operates as a marketplace built on district0x framework. The district0x technology framework is called d0xINFRA. d0xINFRA is in turn powered by Ethereum, Aragon, and IPFS. Learn more at district0x.io. 

Escargot utilizes the four core features of d0xINFRA:

* _Posting & Listing_ - Transacting parties post contracts that include dispute terms and conditions which are bid on by competing arbiters. Alternatively, arbiters post advertisements with fixed price services with blanket terms and conditions, which can be purchased directly by transacting parties. 
* _Search & Filtering_ - Transacting parties can search for arbiters based on reputation, price, terms and conditions, and industry expertise. Arbiters can search for work listed by individuals or businesses seeking escrow services. 
* _Ranking & Reputation_ - Arbiters are evaluated by the community based on decisions previously made on Escargot. Community input determines the reputation of arbiters based on the Escargot Reputation Model. See EgRM below for more details. 
* _Payments & Invoicing_ - Arbiters collect a fee when disputes are raised by transacting parties and resolved on Escargot. Disputes are resolved directly on the Escargot platform and fees are automatically processed. All business executed through Escargot is done using smart contracts that contain a property called simple escrow enabled transaction. See SEET below for more details. 

**2) Escargot Reputation Model (EgRM)** - The collective input from the entire Escargot community must ultimately determine an arbiter’s reputation in the marketplace. In order to faciliate this, the EgRM allows all community members to evaluate past disputes and provide feedback to the arbiters. If the community at large agrees with the decisions of a given arbiter, he will enjoy a strong reputation on the platform. If the community rejects his decisions, his reputation will suffer. Community members are incentivized to participate in EgRM, ensuring a critical mass of careful evaluations. 

EgRM will evolve and grow over time. In future implementations of Escargot, staking of ERC20 tokens may be considered to support an arbiter’s reputation. Private circles may be created to connect smaller, “ultra high quality” networks within the broader community. 

_[ Looking for developers with experience in building reputation systems to join the team or provide feedback on our proposal ]_

**3) Simple Escrow Enabled Transactions (SEET)** - In order for an independent third party to influence the outcome of a transaction on Ethereum, a special type of smart contract must be used. The smart contract must create an temporary escrow which requires subsequent mutual confirmation before funds are finally disbursed. If both sides do not send confirmation, a third party is called upon to decide the outcome of the contract. The decision criteria and possible outcomes that the third party may execute are predefined in the smart contract. 

SEET is a feature of a smart contract rather than a specific contract code. It is a flexible criteria that can be implemented in both simple or complex transaction types. As long as a smart contract is subject to the two-step process using specific parameters, it is considered a SEET.

_[ Looking for developers with experience in smart contract writing to join the team or provide feedback on our proposal ]_

See the diagram below for a breakdown of how the process works:

![eg overview](https://user-images.githubusercontent.com/29905125/28859043-c3cf55da-7709-11e7-9a36-fb1fa171bce7.jpg)

## Real life applications

Escargot is a simple, practical solution that enables commerce transacted on Ethereum today. The service can be adapted for multiple markets and economies. The service is immediately useful for the following cases:

* _Shipment of physical goods (e.g. EthTrade)_ - The most basic type of transaction is between a buyer and seller with a shipment of physical goods. Sellers are often left without recourse if a buyer claims that he never received the purchased item. An escrow service that allows sellers to provide proof of shipment and receipt will enable a truly trustless and decentralized economy. 
* _Freelance employment contracts (e.g. Ethlance)_ - Freelancers rely on employers to approve completed work, but disputes regularly arise over quality standards and perceived completeness. An escrow market that enforces standard terms and specializes in evaluating completed work will enable a fully digital workplace. 
* _Domain sales (e.g. Name Bazaar)_ - Buyers often have no recourse if a domain seller fails to transfer ownership. A simple escrow service will provide much needed security by relying on verifiable digital evidence of ownership transfer.
* _[ Have any other good use cases? Contact us! ]_

## Short term and long term plans

* Today, Escargot is focused on small two-party economic transactions that have low fees and low reputation requirements. While the network is young, we need to focus on building a platform and establishing reputation. The amount of economic risk that transactors are willing to place in Escargot is directly proportional to the collective reputation in the system. **In the future, we service larger, more complex transactions.**

* Today, disputes will be resolved based on simple, digitally-verifiable evidence. Arbiters will be asked to review evidence, primarily in the form of digital documents, publicly available information, and other clearly determinable evidence types. **In the future, complex evidence verification will be permitted such as in person witness and subjective determinations.**

* Today, a single arbiter will make a single decision for each dispute. A simple model for a new economic platform. **In the future, complex arrangements can be created, such as a small council of arbiters that can decide among many outcomes, even split results.**

* Today, Escargot will service other marketplaces on district0x. The success of district0x is dependent on building a strong network effect. Escargot directly services a need for new marketplaces built on district0x. By focusing on servicing its neighbor market first, it will accelerate the network effect, thus benefitting its own long term success. **In the future, Escargot will service economies outside of district0x.**

* Today, reputation will be based entirely on Escargot community input. EgRM will drive a high level of quality into the marketplace, enabling small economic transactions to happen on the platform. As the network scales, it will need to adopt more universally-accepted forms of reputation. **In the future, reputation will consider validation from the broader Ethereum network.**

* Today, transactions and disputes on the Escargot network and publicly executed. This ensures that the entire community is able to influence the arbiter reputation, thus bootstrapping the network effect. **In the future, private networks will be created to service a larger market who demands private interactions, but still requires highly reputable arbiters.**

* _[ Have any other ideas of where we can take this? Contact us! ]_

![screen shot 2017-07-29 at 8 39 43 pm](https://user-images.githubusercontent.com/29905125/28859038-b30559a2-7709-11e7-8a69-7cd112eb6f85.png)

**Comment by cutiebear** [comment]
I like this proposal!

**Comment by joshuam10o** [comment]
Swarm City's smart contracts would be perfect for SEET, because what you are describing is *exactly* what they already do, to some extent. 

Specifically:
https://github.com/swarmcity/sc-dappcontracts/blob/master/contracts/DealForTwo.sol
For an example of a contract that implements something appropriating "SEET"

**Comment by quantumdwayne** [comment]
Great work. District0x definitely needs a service like this. [Ethtrade](https://github.com/district0x/district-proposals/issues/61) would beenfit a lot from Escargot.

**Comment by okmrponcho** [comment]
@joshuam10o Do you have any reference material that explains that script in plain english? 
<--- Not a coder :-)

Or a use case of how it would play out?

**Comment by joshuam10o** [comment]
@okmrponcho, it works almost exactly how you've described Escargot, 

1) User creates a deal, 
2) User specifies the total amount (their part of escrow, plus their counterparty),
3) User funds the deal, (only their part) 
4) User communicates with the counterparty "off chain" 
5) User specifies counterparty to the deal contract. 
6) Counterparty funds the rest. (skin in the game) 
7) Services/goods are provided 
8) User releases deal funds to counterparty. 

At any point before the counterparty is specified, the user can cancel/withdraw 
At any point after the counterparty is specified, but before payout, either party can put the deal into "dispute" 
If the deal is put into dispute, the arbitrator decides what portion of the funding goes to the counterparty, and the rest is refunded to the initial party (this allows any arbitrary distribution). It's implied both parties submit evidence "off chain".

**Comment by ssinsan** [comment]
Hi guys! I also feel like ditrict.Ox would need that kind of service!
Do you know about https://kleros.io/ ?

**Comment by Alexand0x** [comment]
This is the kind of thing I can see being implemented more as a plug-and-play module for all districts to utilize rather than an outright district in its own right. Really useful for bringing value to the entire network!
